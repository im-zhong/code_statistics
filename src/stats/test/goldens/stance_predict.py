#!/usr/bin/env python
# -*- encoding: utf-8 -*-

'''
from xiaohan
这个是立场判断的一个服务

'''

class StanceEnum(str, Enum):
    against = 'AGAINST'
    favor = 'FAVOR'
    none = 'NONE'

if __name__ == "__main__":
    # 此程序只针对中文的处理方法
    print(Stance.get_sentence_stance("happy day"))  # FAVOR
    print(Stance.get_sentence_stance("快乐的一天"))  # None
    print(Stance.get_sentence_stance("你是不是傻"))  # None

''' ''' 

# 我懂了 第一次碰到' 我们就直接认为这是一个字符串了 
# 所以python的字符串判断需要额外进行处理 和block comment分开

from enum import Enum

from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer




    @staticmethod
    def to_name(value):
        for item in StanceEnum:
            if item.value == value:
                return item.name


class Stance:
    """
    Use VaderDict Method
    """
    analyzer = SentimentIntensityAnalyzer()
    lowerb = -0.05
    upperb = 0.05

    @classmethod
    def get_sentence_stance(cls, content: str) -> StanceEnum:
        vs = cls.analyzer.polarity_scores(content)
        compound = vs.pop('compound')
        stance = cls.__infer_stance_with_thresh(compound, lowerb=cls.lowerb, upperb=cls.upperb)
        return stance

    @staticmethod
    def __infer_stance_with_thresh(x, lowerb=-0.05, upperb=0.05) -> StanceEnum:
        if x < lowerb:
            stance = StanceEnum.against
        elif x > upperb:
            stance = StanceEnum.favor
        else:
            stance = StanceEnum.none
        return stance



