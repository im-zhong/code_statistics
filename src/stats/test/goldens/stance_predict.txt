1 : line comment 
2 : line comment 
3 : blank 
4 : block comment 
5 : block comment 
6 : block comment 
7 : block comment 
8 : block comment 
9 : blank 
10 : code 
11 : code 
12 : code 
13 : code 
14 : blank 
15 : code 
16 : line comment 
17 : line comment code 
18 : line comment code 
19 : line comment code 
20 : blank 
21 : block comment 
22 : blank 
23 : line comment 
24 : line comment 
25 : blank 
26 : code 
27 : blank 
28 : code 
29 : blank 
30 : blank 
31 : blank 
32 : blank 
33 : code 
34 : code 
35 : code 
36 : code 
37 : code 
38 : blank 
39 : blank 
40 : code 
41 : block comment 
42 : block comment 
43 : block comment 
44 : code 
45 : code 
46 : code 
47 : blank 
48 : code 
49 : code 
50 : code 
51 : code 
52 : code 
53 : code 
54 : blank 
55 : code 
56 : code 
57 : code 
58 : code 
59 : code 
60 : code 
61 : code 
62 : code 
63 : code 
64 : blank 
65 : blank 
66 : blank 
